pkg/registry/registry.go

```diff
@@ -415,6 +415,9 @@ func (r *registry) newPodTemplateSpec(registryCommand []string, needServiceAccou
 					},
 				},
 			},
+			NodeSelector: map[string]string{
+				"beta.kubernetes.io/os": "linux",
```

Comment:
nit: should these be variables so that the test can just import them?

Reply:
That's a good point - these should be `consts` that are exported